src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -317,7 +317,7 @@ public static string MachineName
 
             // Get the uname's utsname.release.  Then parse it for the first four numbers found.
             // This isn't perfect, but Version already doesn't map exactly to all possible release
-            // formats, e.g. 
+            // formats, e.g. 2.6.19-1.2895.fc6
```

Comment:
Consider factoring out a helper that takes release as a string and adding a unit test

Reply:
All public API must be api-approved, and we avoid exposing internals-visible-to the test assembly. In rare cases we use reflection, I can try that.